mypy/stubtest.py

```diff
@@ -243,6 +243,58 @@ def _belongs_to_runtime(r: types.ModuleType, attr: str) -> bool:
         )
 
 
+IGNORED_DUNDERS = frozenset({
+    # Very special attributes
+    "__weakref__",
+    "__slots__",
+    "__dict__",
+    "__text_signature__",
+    "__match_args__",
```

Comment:
This is probably useful to include in stubs, right? It helps typecheck match statements correctly.

Reply:
Whelp, there are a *lot* of `__match_args__` attributes missing from typeshed. But yes, you make a good point.

Reply:
What's the consensus on what to do with `__match_args__` in a stub file? If the runtime has `__match_args__ = ('foo', 'bar')`, does the stub file have to tediously do `__match_args__: tuple[Literal['foo'], Literal['bar']]`?

Reply:
I hope we can just write `__match_args__ = ('foo', 'bar')`, similar to `__all__`. Haven't seen if that works with type checkers though.

Reply:
I'll experiment locally and see what mypy thinks about it. I'd also support treating it the same way as `__all__`; it seems absurd to do otherwise.

Reply:
Hah, okay.

If I add `__match_args__ = ()` to `_ast.AST` (true), and add `__match_args__ = ('target', 'annotation', 'value', 'simple')`, mypy fails on the stdlib with this message:

```
stdlib\_ast.pyi:87: error: Incompatible types in assignment (expression has type "Tuple[str, str, str, str]", base class "AST" defined the type as "Tuple[]")  [assignment]
```

Reply:
We should fix that (by special-casing `__match_args__`). The way `__match_args__` is used, LSP largely doesn't matter, because it's always looked up on a literal class object. There might already be a bug about it.

Reply:
For now, what do you think we should do?

1. Add the `__match_args__` fields to the `_ast` stub, along with a bajillion `# type: ignore`s?
2. Don't do that, and accept that we may have to add a bajillion allowlist entries if `__match_args__` hasn't been special-cased before the next mypy release?
3. Or leave `__match_args__` on the stubtest blacklist for now, until `__match_args__` has been special-cased by mypy?

Reply:
I think let's leave it on the allowlist for now. Better to fix things one at a time.

Reply:
For completeness: it seems like there already is an open issue about it, and it was filed by a guy called "Jelle Zijlstra": https://github.com/python/mypy/issues/12008

Reply:
who is that